### PR TITLE
Update GameRunner.getChat to return an optional instead of nullable.

### PR DESCRIPTION
**Pull request by DanVanAtta**

### CODE REVIEW

File: src/main/java/games/strategy/engine/framework/GameRunner.java [hunk 1]
```diff
@@ -540,21 +540,16 @@ public static void exitGameIfFinished() {
...
    */
-  public static Chat getChat() {
+  public static Optional<Chat> getChat() {
     final ISetupPanel model = setupPanelModel.getPanel();
     if (model instanceof ServerSetupPanel) {
-      return model.getChatPanel().getChat();
+      return Optional.of(model.getChatPanel().getChat());
     } else if (model instanceof ClientSetupPanel) {
-      return model.getChatPanel().getChat();
+      return Optional.of(model.getChatPanel().getChat());
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/engine/framework/GameRunner.java]: Is there a guarantee that getChat() doesn't return null?

Comment (DanVanAtta) [reply src/main/java/games/strategy/engine/framework/GameRunner.java]: Tracing it down, looks like it should never be null, but we do not have the guarantee. After `shutdown`, for example, chat would be null, but we do not expect it. For now, it's easy enough to use `ofNullable`, I updated that in: 996834792

File: src/main/java/games/strategy/engine/framework/GameRunner.java [hunk 2]
```diff
@@ -540,21 +540,16 @@ public static void exitGameIfFinished() {
...
-      return null;
+      return Optional.empty();
     }
   }
-
-  public static boolean hasChat() {
-    return getChat() != null;
-  }
-
   /**
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/framework/GameRunner.java]: Checkstyle: "'METHOD_DEF' should be separated from previous statement."

File: src/main/java/games/strategy/engine/framework/GameRunner.java [hunk 3]
```diff
@@ -540,21 +540,16 @@ public static void exitGameIfFinished() {
    * todo, replace with something better
    * Get the chat for the game, or null if there is no chat.
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/framework/GameRunner.java]: Javadoc still states `null` may be returned.

File: src/main/java/games/strategy/triplea/ui/TripleAFrame.java [hunk 4]
```diff
@@ -264,15 +264,15 @@ public void windowClosing(final WindowEvent e) {
     chatSplit.setOneTouchExpandable(true);
     chatSplit.setDividerSize(8);
     chatSplit.setResizeWeight(0.95);
-    if (GameRunner.hasChat()) {
+    if (GameRunner.getChat().isPresent()) {
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/ui/TripleAFrame.java]: NB: For future reference, we have an `OptionalUtils#ifPresentOrElse()` "polyfill" for the `Optional#ifPresentOrElse()` method added in Java 9.

File: src/main/java/games/strategy/engine/framework/GameRunner.java (hunk 2 above)
Comment (DanVanAtta) [reply src/main/java/games/strategy/engine/framework/GameRunner.java]: fixed in 826fb08

File: src/main/java/games/strategy/engine/framework/GameRunner.java (hunk 3 above)
Comment (DanVanAtta) [reply src/main/java/games/strategy/engine/framework/GameRunner.java]: Fixed in 47cb6ff

**Pull request merged**